tests/Moq.Analyzers.Test/CallbackAdvancedPatternsValidationTests.cs

```diff
@@ -0,0 +1,318 @@
+            }
+
+            public class TestClass
+            {
+                public void TestMultipleCallbacks()
+                {
+                    {{code}}
+                }
+            }
+            """;
+
+        string source = Template(@namespace, testCode);
+        await AnalyzerVerifier.VerifyAnalyzerAsync(source, referenceAssemblyGroup);
+    }
+
+    [Theory]
+    [MemberData(nameof(DelegateCallbackPatternsData))]
+    public async Task ShouldValidateRefParameterCallbacks(string referenceAssemblyGroup, string @namespace, string testCode)
+    {
+        static string Template(string ns, string mock) =>
+            $$"""
+            {{ns}}
+
+            internal interface IFoo
+            {
+                int DoRef(ref string data);
+                bool DoOut(out int result);
+                string DoIn(in DateTime timestamp);
+            }
+
+            internal class UnitTest
+            {
+                private void Test()
+                {
+                    {{mock}}
+                }
+            }
+            """;
+
+        string source = Template(@namespace, testCode);
+        await AnalyzerVerifier.VerifyAnalyzerAsync(source, referenceAssemblyGroup);
+    }
+
+    [Theory]
+    [MemberData(nameof(RefOutParameterPatternsData))]
+    public async Task ShouldValidateRefOutParameterPatterns(string referenceAssemblyGroup, string @namespace, string testCode)
+    {
+        static string Template(string ns, string code) =>
+            $$"""
+            {{ns}}
+
+            public interface IFoo
+            {
+                int DoWork(string input);
+                bool ProcessMultiple(int id, string name, DateTime timestamp);
+                void ProcessData(ref string data);
+                bool TryProcess(out int result);
+                void ProcessMixed(int id, ref string data, out bool success);
+                void ProcessReadOnly(in DateTime timestamp);
+            }
+
+            public class TestClass
+            {
+                public void TestRefOutCallbacks()
+                {
+                    {{code}}
+                }
+            }
+            """;
+
+        string source = Template(@namespace, testCode);
+        await AnalyzerVerifier.VerifyAnalyzerAsync(source, referenceAssemblyGroup);
+    }
+
+    /// <summary>
+    /// Test to document the current limitation with generic callback validation.
+    /// This test documents that .Callback&lt;T&gt;() with wrong type parameters is NOT currently validated.
+    /// This could be enhanced in a future version.
+    /// </summary>
+    /// <returns>A task representing the asynchronous unit test.</returns>
+    [Fact]
+    public async Task GenericCallbackValidation_CurrentLimitation_IsDocumented()
+    {
+        const string source = """
+            using Moq;
+
+            public interface IFoo
+            {
+                int DoWork(string input);
```

Comment:
This is just a subset of the template in `ShouldValidateRefOutParameterPatterns` and can be reused. Make this data driven and combine where there's overlapping templates